.travis.yml

```diff
@@ -0,0 +1,38 @@
+language: cpp
+compiler: gcc
+dist: xenial
+addons:
+  apt:
+    packages: lcov
+
+# C++ 17 install from:
+# https://github.com/richelbilderbeek/travis_gcc_cpp17/blob/master/.travis.yml
+
+before_install:
+    # C++17
+    - sudo add-apt-repository -y ppa:ubuntu-toolchain-r/test
+    - sudo apt-get update -qq
+    - pip install --user cpp-coveralls
+
+install:
+    - sudo apt-get install -qq libopencv-dev
+    # C++17
+    - sudo apt-get install -qq g++-6
+    - sudo update-alternatives --install /usr/bin/g++ g++ /usr/bin/g++-6 90
+
+script:
+    - mkdir build
+    - cd build
+
+    - cmake -DUSE_GCOV=yes ..
```

Comment:
This compiles the tests with g++ optimization-level '-O0', but in production it will always be '-O3' or '-O1'. I think such differences should be avoided.

Reply:
When GCOV is so important, cant we have two builds? One for running the tests and a second one for GCOV with '-O0'?